### PR TITLE
[common] - add ingressclass name

### PR DESCRIPTION
**Pull request by angryninja48**
Signed-off-by: Jon Baker <jonbaker85@gmail.com>

**Description of the change**

Adds in ingressclass name to the spec field

https://kubernetes.io/blog/2020/04/02/improvements-to-the-ingress-api-in-kubernetes-1.18/

**Benefits**

Able to use ingress class in K8 >= 1.18 

**Possible drawbacks**

None

**Applicable issues**

- fixes #

**Additional information**

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Title of the PR starts with chart name (e.g. `[home-assistant]`)
- [ ] Chart is using our [common](https://github.com/k8s-at-home/charts/tree/master/charts/common) library as a dependency.
- [ ] (optional) Variables are documented in the README.md

### CODE REVIEW

**Comment by onedr0p** [comment]
You will need to make the logic compare kubernetes versions because it breaks linting. 

See this comment
https://github.com/k8s-at-home/charts/blob/master/charts/common/templates/lib/chart/_capabilities.tpl#L81

**Comment by onedr0p** [comment]
Additionally you _should_ be able to use this following logic for that field, right?

https://github.com/k8s-at-home/charts/blob/master/charts/common/templates/classes/_ingress.tpl#L46

File: charts/common/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v2
 name: common
 description: Function library for k8s-at-home charts
 type: library
-version: 2.1.1
+version: 2.2.1
```
Comment (onedr0p) [inline charts/common/Chart.yaml:5]: This should be 2.2.0

File: charts/common/templates/classes/_ingress.tpl [hunk 2]
```diff
@@ -26,6 +26,11 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  {{- if or (.Capabilities.APIVersions.Has "networking.k8s.io/v1/IngressClass") (.Capabilities.APIVersions.Has "networking.k8s.io/v1beta1/IngressClass") }}
```
Comment (onedr0p) [inline charts/common/templates/classes/_ingress.tpl:29]: See other comment in PR

**Review by onedr0p — CHANGES_REQUESTED** [review]
Small changes

**Review by angryninja48 — COMMENTED** [review]
Updated to follow existing logic

**Comment by bjw-s** [comment]
Thanks for the contribution!
I have changed the base branch to `common-next` so that we can group a few changes together for the next release of the `common` chart.

**Review by bjw-s — APPROVED** [review]

**Pull request merged**